[Derrick Teo] Duke Increments

teoha opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

teoha committed:
Level-1

teoha committed:
Level-2

teoha committed:
Edited switch to include breaks

teoha committed:
Level-3

teoha committed:
Added prompts to Level-3

teoha committed:
Level-4

teoha committed:
Level-5

teoha committed:
Removed method TasksDeleted

teoha committed:
Level-6

teoha committed:
Added UI testing

teoha committed:
Level-7

teoha committed:
Level-8

teoha committed:
Merge branch 'branch-Level-7'

teoha committed:
Merge branch 'branch-Level-8'

teoha committed:
Fixed typo in format specifier

teoha committed:
A-MoreOOP

teoha committed:
A-Packages

teoha committed:
Fixed error in day to date

teoha committed:
A-JUnit

teoha committed:
Added v0.1 release manifest file

teoha committed:
Merge branch 'gradle' of https://github.com/teoha/duke

teoha committed:
A-JavaDoc

teoha committed:
A-CodingStandard

teoha committed:
Level-9

teoha committed:
Merge branch 'branch-A-JavaDoc'

teoha committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/AddDeadlineCommand.java
#	src/main/java/duke/command/AddEventCommand.java
#	src/main/java/duke/command/AddTodoCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/DoneCommand.java
#	src/main/java/duke/command/ExitCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/exception/EmptyDescriptionException.java
#	src/main/java/duke/exception/InvalidCommandException.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/ToDo.java

teoha committed:
Fixed typo in javadoc "IOException"

teoha committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java

teoha committed:
Fixed all checkstyle errors and bug in parsing storage

teoha committed:
Fixed storage parsing for Deadline

teoha committed:
Refactored storage parsing into Parse class

teoha committed:
Implemented a GUI

teoha committed:
Gradle set up

teoha committed:
Set up gradle checkstyle and fixed errors

teoha committed:
Merge branch 'branch-A-CheckStyle'

teoha committed:
Set up Gradle

teoha committed:
Merge branch 'branch-A-Gradle'

teoha committed:
Merge branch 'branch-Level-10'

limerencee commented on src/main/java/duke/Storage.java:
```diff
@@ -55,6 +55,7 @@ public Storage(String filePath) {
     public void saveTodo(String description) throws IOException {
```
You could consider combining your `saveToDo(), saveEvent() and saveDeadline()` into a single method. For example:
```
public void saveToFile(String ... tokens) throws IOException {
    if (tokens.length == 1) {
        fileWriter.write(...); //todo
    } else {
        if (tokens[0].equals("event") {
            fileWriter.write(...); //event
        } else if (tokens[0].equals("deadline") {
            fileWriter.write(...); //deadline
        }
    }
}
```

limerencee commented on src/main/java/duke/Parser.java:
```diff
@@ -16,6 +16,9 @@ public static Command parse(String fullCommand) throws EmptyDescriptionException
             case ("done"):
                 int index = Integer.parseInt(fullCommand.split(" ")[1]) - 1;
```
Should `catch` or `throw` `NumberFormatException` if the input is non-numerical.

limerencee commented on src/main/java/duke/Parser.java:
```diff
@@ -16,6 +16,9 @@ public static Command parse(String fullCommand) throws EmptyDescriptionException
                 int index = Integer.parseInt(fullCommand.split(" ")[1]) - 1;
                 return new DoneCommand(index);
+            case("find"):
+                String keyword = strArr[1].trim();
```
For a possible additional feature, allow for multi-word searching. This can only search for a single word/letter. One method would be to concatenate the remainder of the array elements in your `strArr`.

limerencee commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,39 @@
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException, EmptyDescriptionException, ParseException {
+        List<Task> relevantList = new ArrayList<>();
+        for(int i=0;i<tasks.listSize(); i++){
```
Coding style: consider adding spaces in your parenthesis arguments.
`for (int i = 0; i < tasks.listSize(); i++) {`

limerencee replied:
Make use of IDE to refactor your code to comply to the coding style

limerencee commented on src/main/java/duke/Parser.java:
```diff
@@ -1,3 +1,9 @@
+package duke;
+
+import duke.command.*;
```
Coding style: Avoid wildcard imports!

limerencee commented on src/main/java/Event.java:
```diff
@@ -1,13 +1,17 @@
+    private Date end;
 
-    public Event(String description, String duration) {
+    public Event(String description, Date start, Date end) {
```
I like how you improved on the Event type, by changing the description to include Start and End Date!

limerencee commented on src/main/java/Duke.java:
```diff
@@ -1,13 +1,47 @@
+        FileWriter fileWriter=null;
+
+        try {
+            File file = new File("./data/duke.txt");
```
Consider declaring and using `final String` for hard-coded Strings for ease of future modifications (if necessary).

teoha committed:
Updated GUI

teoha committed:
Added assertions checking for storage date validity

teoha committed:
Merge pull request #3 from teoha/branch-A-Assertions

Added assertions checking for storage date validity. Merged using regular merge.

teoha committed:
Merge branch 'master' of https://github.com/teoha/duke

teoha committed:
Fixed several CheckStyle errors

teoha committed:
Branch a code quality (#4)

* Fixed several CheckStyle errors

* Set up PMD gradle plugin to check code quality and fixed several issues found

* Testing squash and merge pull request option

teoha committed:
Merge branch 'master' of https://github.com/teoha/duke

# Conflicts:
#	src/main/java/duke/util/Parser.java

teoha committed:
Added .travis.yml to enable continuous integration

teoha committed:
Permission access for travis

teoha committed:
Set the line endings of gradlew back to Unix style

teoha committed:
Temporarily removed pmd to test travis build

teoha committed:
1. Revamped storage strategy
2. Added test for TaskList
3. Updated JavaDocs for better clarity
4. Completed C-Priority

teoha committed:
Fixed bug where months were displayed incorrectly. Added screenshot of product.

teoha committed:
Added user guide and fixed a bug where delete deletes the wrong task

teoha committed:
Fixed screenshot link and syntax of the readme

teoha committed:
Cleaned up Javadocs

teoha committed:
Fixed file format in readme

teoha committed:
Fixed several checkstyle errors and changed gradle's Main class to Launcher

teoha committed:
Added cross platform support

teoha committed:
Applied SLAP to most classes

teoha committed:
Attempt to fix .PNG file extension

teoha committed:
Deleted

teoha committed:
Added

teoha committed:
updated README to use .png extension to identify UI.png
